dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2639,325 @@ void FileSprayer::pushParts()
+
+        return true;
+    }
+    static void buildAzureStorageURI(IDistributedFile *distFile, unsigned partNum, const char * baseURI, const char *sasToken, StringBuffer & uri)
+    {
+        StringBuffer path;
+        bool ret = distFile->getUnmountedPath(partNum, path);
+        assertex(ret);
+        StringBuffer tmp;
+        uri.appendf("%s/%s%s", baseURI, encode_url(tmp, path.str()).str(), sasToken);
+    }
+};
+
+class AzureFileClient : public CApiCopyClient
+{
+    std::unique_ptr<Shares::ShareFileClient> fileClient;
+    std::unique_ptr<Shares::StartFileCopyOperation> fileCopyOp;
+public:
+    ApiCopyStatus startCopy(const char *target, const char * source)
+    {
+        try
+        {
+            fileClient.reset(new Shares::ShareFileClient(target));
+            fileCopyOp = std::make_unique<Shares::StartFileCopyOperation>(std::move(fileClient->StartCopy(source)));
+        }
+        catch (const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("AzureFileClient startCopy failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
+        }
+        status = ApiCopyStatus::Pending;
+        return status;
+    }
+    ApiCopyStatus getProgress(CDateTime & dateTime, int64_t & outputLength)
+    {
+        try
+        {
+            fileCopyOp->Poll();
+            Shares::Models::FileProperties props = fileCopyOp->Value();
+            dateTime.setString(props.LastModified.ToString().c_str());
+            outputLength = props.FileSize;
+            Shares::Models::CopyStatus tstatus = props.CopyStatus.HasValue()?props.CopyStatus.Value():(Shares::Models::CopyStatus::Pending);
+            if (tstatus==Shares::Models::CopyStatus::Success) // FYI. CopyStatus is an object so can't use switch statement
+                status = ApiCopyStatus::Success;
+            else if (tstatus==Shares::Models::CopyStatus::Pending)
+                status = ApiCopyStatus::Pending;
+            else if (tstatus==Shares::Models::CopyStatus::Aborted)
+                status = ApiCopyStatus::Aborted;
+            else
+                status = ApiCopyStatus::Failed;
+        }
+        catch(const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("Transfer using API .Poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
+        }
+        return status;
+    }
+    ApiCopyStatus abortCopy()
+    {
+        try
+        {
+            fileCopyOp->Poll();
+            Shares::Models::FileProperties props = fileCopyOp->Value();
+            if (props.CopyStatus.HasValue() && props.CopyStatus.Value()!=Shares::Models::CopyStatus::Success)
+            {
+                if (fileCopyOp->HasValue() && fileCopyOp->Value().CopyId.HasValue())
+                    fileClient->AbortCopy(fileCopyOp->Value().CopyId.Value().c_str());
```

Comment:
Should you also Delete the blob?
Should you check the return (and log if the abort fails for some reason)

Reply:
I'll change to delete blob if abort called and report the return status of the .AbortCopy call.